sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/FlattenTest.java

```diff
@@ -228,7 +228,7 @@ public void testFlattenPCollectionsEmptyThenParDo() {
   public void testFlattenNoListsNoCoder() {
     // not ValidatesRunner because it should fail at pipeline construction time anyhow.
     thrown.expect(IllegalStateException.class);
-    thrown.expectMessage("cannot provide a Coder for empty");
+    thrown.expectMessage("Unable to return a default Coder");
 
     PCollectionList.<ClassWithoutCoder>empty(p)
```

Comment:
I suppose cases like this can be handled by having transforms such as empty Create and empty Flatten set a special Coder that says it can encode anything, but actually will throw when asked to encode anything. Likewise, WithKeys.of(null) or Create.of(null) could use a "null only" coder. That'd be a nice cleanup but outside the scope of this PR.